### PR TITLE
Make ambassador cards shiny

### PR DESCRIPTION
**Pull request by farrenr**
Resolves [#67](https://github.com/alveusgg/extension/issues/67)

- Used [react-parallax-tilt](https://www.npmjs.com/package/react-parallax-tilt) to create tilt and shine effects on ambassador cards.
- Added respective user setting to disable card effects.

### CODE REVIEW

File: src/components/ambassadorCard/AmbassadorCard.tsx [hunk 1]
```diff
@@ -30,142 +32,156 @@ export interface AmbassadorCardProps {
...
-      )}
+    <Tilt
+      tiltEnable={!disableCardEffects}
+      glareEnable={!disableCardEffects}
+      glareMaxOpacity={glareOpacity}
+      glareBorderRadius="1rem"
+      glarePosition="bottom"
+      scale={1.0}
+      perspective={5000}
+      className={classes(styles.ambassadorCard, className)}
```
Comment (MattIPv4) [inline src/components/ambassadorCard/AmbassadorCard.tsx:56]: Unfortunately something here sets a transform-origin that makes this perspective shift look rather funky. I wonder if we need to move that transform-origin (which is used for the slide-in/out animation) into a wrapper around the Tilt component?

File: src/components/ambassadorCard/AmbassadorCard.tsx [hunk 2]
```diff
@@ -30,142 +32,156 @@ export interface AmbassadorCardProps {
...
-              &times;
-            </button>
-          )}
-
-          <h2 className={styles.name} title={ambassador.name}>
-            {ambassador.name}
-          </h2>
+      <div
+        className={classes(
+          ambassador.birth && isBirthday(ambassador.birth) && styles.birthday,
```
Comment (MattIPv4) [inline src/components/ambassadorCard/AmbassadorCard.tsx:60]: The birthday hat looks to be broken?

File: src/components/ambassadorCard/AmbassadorCard.tsx [hunk 3]
```diff
@@ -30,142 +32,156 @@ export interface AmbassadorCardProps {
...
-        ambassador.birth && isBirthday(ambassador.birth) && styles.birthday,
-      )}
+    <Tilt
+      tiltEnable={!disableCardEffects}
+      glareEnable={!disableCardEffects}
+      glareMaxOpacity={glareOpacity}
+      glareBorderRadius="1rem"
+      glarePosition="bottom"
+      scale={1.0}
+      perspective={5000}
```
Comment (MattIPv4) [inline src/components/ambassadorCard/AmbassadorCard.tsx:55]:
```suggestion
      perspective={7500}
      tiltReverse
```
From playing around with the settings locally, this felt nicer to me?

File: src/components/ambassadorCard/AmbassadorCard.tsx [hunk 4]
```diff
@@ -30,142 +32,156 @@ export interface AmbassadorCardProps {
...
-          )}
-
-          <h2 className={styles.name} title={ambassador.name}>
-            {ambassador.name}
-          </h2>
+      <div
+        className={classes(
+          ambassador.birth && isBirthday(ambassador.birth) && styles.birthday,
+        )}
+      >
```
Comment (MattIPv4) [inline src/components/ambassadorCard/AmbassadorCard.tsx:62]: It looks like this extra inner div has also broken scrolling on the content?

**Review by MattIPv4 — CHANGES_REQUESTED** [review]
👋 Looking great, but there are a few issues I noticed:

File: src/components/ambassadorCard/AmbassadorCard.tsx (hunk 1 above)
Comment (farrenr) [reply src/components/ambassadorCard/AmbassadorCard.tsx:56]: Ah I wasn't sure what was happening here, thanks for the spot - I'll take a look!

File: src/components/ambassadorCard/AmbassadorCard.tsx (hunk 4 above)
Comment (farrenr) [reply src/components/ambassadorCard/AmbassadorCard.tsx:62]: Not sure what happened here, will take a look at this and the birthday hat

File: src/components/ambassadorCard/AmbassadorCard.tsx (hunk 1 above)
Comment (farrenr) [reply src/components/ambassadorCard/AmbassadorCard.tsx]: When investigating this, I trialled removing the transform-origin from the ambassador card class within the overlay scss module and this appears to achieve the desired result. The cards still slide in from left and now the tilt works correctly. Let me know what you think, as far as I could tell, removing the transform-origin hasn't had any unwanted side effects!

File: src/components/ambassadorCard/AmbassadorCard.tsx (hunk 3 above)
Comment (farrenr) [reply src/components/ambassadorCard/AmbassadorCard.tsx]: Agree! Change made

File: src/components/ambassadorCard/AmbassadorCard.tsx (hunk 4 above)
Comment (farrenr) [reply src/components/ambassadorCard/AmbassadorCard.tsx]: Inner div removed as we now have the Tilt component div.

File: src/components/ambassadorCard/AmbassadorCard.tsx [hunk 5]
```diff
@@ -30,17 +32,27 @@ export interface AmbassadorCardProps {
...
   return (
-    <div
+    <Tilt
+      tiltEnable={!disableCardEffects}
+      glareEnable={!disableCardEffects}
+      glareMaxOpacity={glareOpacity}
+      glareBorderRadius="1rem"
+      glarePosition="bottom"
+      scale={1.0}
+      perspective={2500}
```
Comment (MattIPv4) [inline src/components/ambassadorCard/AmbassadorCard.tsx:55]:
```suggestion
      perspective={2500}
      tiltReverse
```
Can we enable `tiltReverse`? It feels more natural to me?

File: src/components/ambassadorCard/AmbassadorCard.tsx
Comment (MattIPv4) [inline src/components/ambassadorCard/AmbassadorCard.tsx:1]: The only remaining issue looks to be that the tooltip on conservation status is now showing in the middle of nowhere?

Comment (farrenr) [reply src/components/ambassadorCard/AmbassadorCard.tsx:1]: Good spot! Let me take a look

Comment (MattIPv4) [reply src/components/ambassadorCard/AmbassadorCard.tsx:1]: 👋 @farrenr checking in if you had a chance to look at this?

Comment (farrenr) [reply src/components/ambassadorCard/AmbassadorCard.tsx:1]: Hey! Apologies on taking a while with this, I've been a bit stuck on the last issue with the tooltip but I'm looking at it today

**Comment by pjeweb** [comment]
I think it looks very cool, but I think it could be a bit smoother. The problem with it resetting on the edges and flickering seems to be inherent to the library: https://imgur.com/6nkHLFe

I think with increasing the transition speed to like 5s or so it gets a lot smoother overall, especially when leaving the card as it does not "snap" back to quickly. 

I also think the glare could be less or the font contrast gets too low. Its already not the best contrast. Maybe reducing the opacity to 30% could work.

Another thing i noticed, we do have a checkbox to disable the effect. Is there any way to also disable it when `prefers-reduced-motion` is active? Not sure how to handle the two - maybe just make the checkbox readonly with a note ("disabled due to browser preferences") or hide the checkbox completely?

**Comment by MattIPv4** [comment]
👋 Given the lack of movement on this, and the conflicts (mostly my doing, sorry), I'm going to close this for now. Please feel free to open a new PR with updates if/when you have the time.

**Pull request closed without merging**